core/activity_pub.md

```diff
@@ -6,35 +6,46 @@ This document specifies the extensions to Fora required for a community to be co
 
 
-## Public/Private Key Authentication
+## URI and DID
 
-ActivityPub does not specify a specific authentication scheme, however most ActivityPub communities use OAuth2.0 or something similar which involves a traditional username/password scheme. Since all of the community state must be public and shareable between hosts, we cannot rely on username/password schemes or any scheme that gives hosts privileged information on a user’s authentication info. Thus, users must authenticate by proving ownership of a secret private key that corresponds to the public key that is publicly registered on a blockchain.
+URIs (Uniform Resource Identifier) are used within Activity Pub to identify different servers and clients.  
 
-## URI and DID
+Fora will use [decentralized identifiers](https://www.w3.org/TR/did-core/), to identify both clients and communities. 
 
-URIs (Uniform Resource Identifier) are used within Activity Pub to identify different servers. 
+Please refer to the DID specification for further details, but it is essential to understand that:
+- a DID identifies a DID Subject, and that subject must never change
+- a DID resolves to a DID Document which describes the DID subject
+- a DID Document may be modified by one or more DID Controllers
+- a DID Subject is not necessarily also the DID Controller
+- there exists a verification method for each DID controller, which is specified in the DID Document
+- there exists an authentication method to be associated with a DID Subject, which is specified in the DID Document
+- All information may be modified in a DID Document except the id of the subject
 
-Fora will use [decentralized identifiers](https://www.w3.org/TR/did-core/), to identify both users and platforms. These DIDs must be registered on a blockchain that the platform can query identities from. Users will have a DID so they can self-attest to their identity on the blockchain and then use that DID in any Fora community they wish. They can also retain this identity even as they move to different communities. The DID document for each user must contain the current public key they will use sign messages sent from their Fora user account. 
+The DIDs of a client must be recorded on a verifiable data registry, typically this will be a public blockchain or ledger. All communities will rely on client DIDs to resolve the authentication method used by a particular client. Communities may choose not to rely on DIDs for authentication of governance module interaction (NOTE: the authentication method of a client may be different from its corresponding controller, the latter would be used to authenticate a governance vote). In this case, public keys for each client controller must be registered on the community. If the communities chooses to use on-chain DID verification, the client DID controllers must be verifiable via a communication protocol such as InterBlockchain Communication.
```

Comment:
"(NOTE: the authentication method of a client may be different from its corresponding controller, the latter would be used to authenticate a governance vote)"

Could you clarify what you mean here? Do you mean that the PubKey used to send ActivityPub messages could be different than the pubkey used to send governance votes

Reply:
Correct. There exists [authorization](https://www.w3.org/TR/did-core/#did-controller) in a DID Document
```
Authorization is the mechanism used to state how operations are performed on behalf of the DID subject. A DID controller is authorized to make changes to the respective DID document. 
```

As well as [authentication](https://www.w3.org/TR/did-core/#authentication). 
```
The authentication verification relationship is used to specify how the DID subject is expected to be authenticated, such as for the purposes of logging into a website. 
```

On further thought, we could just use `authentication` as the method for allowing voting permissions. The controllers could always change the authentication method. 

My concern here is for complex cases, say if a member is actually another fora network. In this case, there may exist a slew of authentication methods (public keys for each sub-member), but authorization might be a multi-sig of the sub-members or even a passed gov proposal. Here authorization is a stronger property than authentication and if the network is to vote, you likely want the strongest verification method?

Thoughts?

Reply:
Let's do this. Ideally, members have a DID that defines a third verification relationship `voting authentication` or maybe `blockchain authentication`. In this situation, we use that verification relationship for verifying votes. In this verification relationship does not exist (member DIDs could use entirely different DID implementations), then we should gracefully fail on using the authorization method of the DID since I'm pretty sure that should always be there

Reply:
see pushed changes

Reply:
I'll put up a new pr later, but looks like we can use capabilities. It's unclear to me rn if you can create multiple capabilities with various delegates for each capability
```
 The capabilityDelegation verification relationship is used to specify a mechanism that might be used by the DID subject to delegate a cryptographic capability to another party, such as delegating the authority to access a specific HTTP API to a subordinate.

```